### PR TITLE
Bluebutton-983: new code to handle hicn pointing to more than one bene

### PR DESCRIPTION
**Pull request by dho937**
@karlmdavis   Can I get a review on this ASAP?  This is part of the code I need to move to production soon to fix the 500 errors we are getting where a HICN points to more than one bene id.  The JUnit I created is a little complex but don't have time to refactor now as this needs to get to production soon.  Thanks.

### CODE REVIEW

File: bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/PatientResourceProvider.java [hunk 1]
```diff
@@ -305,4 +346,50 @@ private Patient queryDatabaseByHicnHash(String hicnHash) {
 		Patient patient = BeneficiaryTransformer.transform(metricRegistry, beneficiary);
 		return patient;
 	}
+
+	/**
+	 * 
+	 * Following method will bring back the bene id that has the most recent
+	 * rfrnc_yr since the hicn points to more than one bene id in the
+	 * Beneficiaries table
```
Comment (karlmdavis) [inline bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/PatientResourceProvider.java]: How do we know that this the right thing to do in this circumstance?

Also: are there any cases where a single HICN points to _multiple_ bene records with the same `RFRNC_YR`? If so this code won't notice; it'll just retrurn the first of those that it happens to come across.

**Review by karlmdavis — COMMENTED** [review]
@dho937 Great work!

I noted a couple of important questions in a comment on your `getMaxReferenceYearForDupBenes(...)` method. We'll need to get those answers if we don't have them, and document them, e.g. in a comment.

Also: given the changes I made yesterday in https://github.com/CMSgov/bluebutton-data-server/pull/157, this will need a fair bit of rework, but should also end up much simpler (as there's a spot in the code where we already have all of the matching bene records and their data available, so you won't need to run any queries).

**Review by jcgeer — APPROVED** [review]
Looks good to me, but I don't see anywhere you've answered Karl's questions concerning always returning the bene with the latest `RFRNC_YR`.

**Comment by karlmdavis** [comment]
Closing this PR, as the project has now been migrated into a new monorepo: https://github.com/CMSgov/beneficiary-fhir-data

**Pull request closed without merging**